### PR TITLE
Add map math mixin

### PR DESCRIPTION
**Pull request by alexander-schranz**
**Map math**

Usage:

```scss
mapMathHalf((
     desktop: 100px,
     mobile: 50px
), 1.2);
```

Output:

```scss
(
    desktop: 120px,
    mobile: 60px
)
```

**Map math half**

Usage:

```scss
mapMathHalf(
     desktop: 100px,
     mobile: 50px
);
```

Output:

```scss
(
    desktop: 50px,
    mobile: 20px
)
```

**Map math negative**

Map math negative will invert the given values.

Usage:

```scss
mapMathNegative(
    desktop: 100px,
    mobile: 50px
);
```

Output:

```scss
(
    desktop: -100px,
    mobile: -50px
)
```

**Example Usage with mediaEachMax**:

```scss
$gutter-widths: (
    default: 22px,
    desktop: 18px,
    tablet: 16px,
    smart: 6px,
);

.container {
    @include mediaEachMax(mapMathHalfNegative($gutter-widths), (
        margin-left,
        margin-right
    ));
}

.container__item {
     float: left;
     width: 25%;

    @include mediaEachMax(mapMathHalf($grid-gutter-widths), (
        padding-left,
        padding-right
    ));
}
```

Output:

```sass
.container {
  margin-left: -11px;
  margin-right: -11px;
}

@media only screen and (max-width: 1199px) {
  .container {
    margin-left: -9px;
    margin-right: -9px;
  }
}

@media only screen and (max-width: 991px) {
  .container {
    margin-left: -8px;
    margin-right: -8px;
  }
}

@media only screen and (max-width: 767px) {
  .container {
    margin-left: -3px;
    margin-right: -3px;
  }
}

.container__item {
  float: left;
  width: 25%;
  padding-left: 11px;
  padding-right: 11px;
}

@media only screen and (max-width: 1199px) {
  .container__item {
    padding-left: 9px;
    padding-right: 9px;
  }
}

@media only screen and (max-width: 991px) {
  .container__item {
    padding-left: 8px;
    padding-right: 8px;
  }
}

@media only screen and (max-width: 767px) {
  .container__item {
    padding-left: 3px;
    padding-right: 3px;
  }
}
```

### CODE REVIEW

File: scss/tools/_map-math.scss [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @return $newMap;
+}
+
+// Map math negative will invert the given values and divide by 2.
+//
+// Usage:
+//
+// mapMathNegative(
```
Comment (c00n84) [inline scss/tools/_map-math.scss]: @alexander-schranz check comment. Should be `mapMathHalfNegative`.

**Comment by c00n84** [comment]
else ACK

File: scss/tools/_map-math.scss (hunk 1 above)
Comment (alexander-schranz) [reply scss/tools/_map-math.scss]: fixed

**Comment by alexander-schranz** [comment]
needs: https://github.com/massiveart/stylelint-config-ma/pull/3

**Pull request merged**